### PR TITLE
feat(generators): add register in index + create settings json

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

When you use the generator in a project and you add a component
You will have to register that component in src/app/component/index.js
and you will maybe add a settings json for it.

**What is the chosen solution to this problem?**

add the following questions to the generator of components:

* register into src/app/component/index.js
* add src/settings/MyComponent.json

Bonus:

* Add missing displayName in the existing components
* Remove useless mapStateToProps example
* Remove the useless arrow function component

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/generator/generators/react-component/index.js [hunk 1]
```diff
@@ -106,5 +130,49 @@ module.exports = yeoman.Base.extend({
...
+		if (this.props.parentIndex) {
+			const indexPath = path.join(this.props.path, './index.js');
+			let code;
+			try {
+				code = fs.readFileSync(indexPath);
+			} catch (error) {
+				this.log(error.message);
+			}
+			if (code) {
+				const ast = recast.parse(code);
```
Comment (smouillour) [inline packages/generator/generators/react-component/index.js]: Could you change the variable's name to be more explicit?

File: packages/generator/generators/react-component/index.js [hunk 2]
```diff
@@ -55,9 +58,30 @@ module.exports = yeoman.Base.extend({
 			name: 'path',
 			message: 'path',
 			default: 'src/app/components',
+		}, {
+			type: 'input',
+			name: 'settings',
+			message(a) {
```
Comment (smouillour) [inline packages/generator/generators/react-component/index.js]: what is `a`?

File: packages/generator/generators/react-component/index.js [hunk 3]
```diff
@@ -106,5 +130,49 @@ module.exports = yeoman.Base.extend({
...
+			try {
+				code = fs.readFileSync(indexPath);
+			} catch (error) {
+				this.log(error.message);
+			}
+			if (code) {
+				const ast = recast.parse(code);
+				const b = recast.types.builders;
+				// first find last import
+				const imports = ast.program.body.filter(item => item.type === 'ImportDeclaration');
```
Comment (smouillour) [inline packages/generator/generators/react-component/index.js]: that's don't need any verification on `ast` before use `filter`?

File: packages/generator/generators/react-component/index.js [hunk 4]
```diff
@@ -106,5 +130,49 @@ module.exports = yeoman.Base.extend({
...
+				const lastIndexImport = ast.program.body.indexOf(imports[imports.length - 1]);
+				// inject the new import after the last one
+				ast.program.body.splice(lastIndexImport + 1, 0, b.importDeclaration([
```
Comment (smouillour) [inline packages/generator/generators/react-component/index.js]: same for slice?

File: packages/generator/generators/react-component/index.js (hunk 1 above)
Comment (jdescottes-talend) [reply packages/generator/generators/react-component/index.js]: mostly `b` below

File: packages/generator/generators/react-component/index.js [hunk 5]
```diff
@@ -106,5 +130,49 @@ module.exports = yeoman.Base.extend({
...
+				code = fs.readFileSync(indexPath);
+			} catch (error) {
+				this.log(error.message);
+			}
+			if (code) {
+				const ast = recast.parse(code);
+				const b = recast.types.builders;
+				// first find last import
+				const imports = ast.program.body.filter(item => item.type === 'ImportDeclaration');
+				const lastIndexImport = ast.program.body.indexOf(imports[imports.length - 1]);
```
Comment (jdescottes-talend) [inline packages/generator/generators/react-component/index.js]: please don't make computation inside function call, extract the `lastImport` in a variable

File: packages/generator/generators/react-component/index.js (hunk 4 above)
Comment (jdescottes-talend) [inline packages/generator/generators/react-component/index.js]: same here, define the new import declaration inside a variable.

File: packages/generator/generators/react-component/index.js (hunk 1 above)
Comment (jmfrancois) [reply packages/generator/generators/react-component/index.js]: parsedIndex ?

**Pull request merged**